packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -76,19 +76,30 @@ export interface CheckoutEvents {
 	/**
 	 * {@inheritdoc TreeViewEvents.commitApplied}
 	 */
-	commitApplied(data: CommitMetadata, getRevertible?: () => Revertible): void;
-
-	/**
-	 * Fired when a revertible is either reverted or discarded.
-	 *
-	 * This event can be used to maintain a list or set of active revertibles.
-	 * @param revertible - The revertible that was disposed.
-	 * This revertible was previously passed to the `newRevertible` event.
-	 * Calling `discard` on this revertible is not necessary but is safe to do.
-	 */
-	revertibleDisposed(revertible: Revertible): void;
+	commitApplied(data: CommitMetadata, getRevertible?: RevertibleFactory): void;
 }
 
+/**
+ * Factory for creating a {@link Revertible}.
+ *
+ * @param onRevertibleDisposed - A callback that will be invoked when the `Revertible` generated by this factory is disposed.
+ * Note that `SharedTree` retains the right to unilaterally dispose of revertibles. This happens when the view that
+ * the revertible was generated on is disposed.
+ *
+ * @public
+ */
+export type RevertibleFactory = (onRevertibleDisposed?: RevertibleDisposedCallback) => Revertible;
+
+/**
+ * A callback invoked after the disposal of a {@link Revertible}.
+ * This is typically used to clean up any resources associated with the `Revertible` in host application.
+ *
+ * @param disposed - the `Revertible` that was disposed.
+ *
+ * @public
+ */
+export type RevertibleDisposedCallback = (disposed: Revertible) => void;
```

Comment:
nit: could rename `disposed` to `revertible` - disposed sounds like a bool.

Though, IMO I would just inline this type in `RevertibleFactory`. I think that:

`(onRevertibleDisposed?: (revertible: Revertible) => void)`

is self-explanatory and doesn't need any additional naming or documentation.

Reply:
I find that naming a `Revertible` "revertible" doesn't add much value, and a boolean ought to be `isDisposed`. That said, I think it's pretty clear what's happening so "revertible"" is fine.

Reply:
You make a good point. Ideally we wouldn't need to name it at all, but if we have to, then we can at least add slightly more meaning/clarity by calling it "disposed" rather than "revertible" which adds no additional meaning at all. I agree with that, but I still think "disposed" sounds kind of like a bool, so it's a wash. "disposedRevertible" is too long. Okay, I've got it - have the function take no arguments, but bind it to the revertible before invoking so that the user can get the revertible via `this` ^.^ EZ PZ

In seriousness though, call it whatever you prefer, I think `disposed` and `revertible` are both acceptable.